afs/afs-core/src/main/java/com/powsybl/afs/AppData.java

```diff
@@ -201,13 +172,13 @@ public ListenableAppStorage getRemotelyAccessibleStorage(String fileSystemName)
         return afs != null ? afs.getStorage() : null;
     }
 
-    <U> U findService(Class<U> serviceClass, boolean remote) {
+    <T extends ProjectFile, U> ServiceExtension<T, U> findService(Class<U> serviceClass, boolean remote) {
         ServiceExtension.ServiceKey<U> serviceKey = new ServiceExtension.ServiceKey<>(serviceClass, remote);
-        U service = (U) services.get(serviceKey);
-        if (service == null) {
-            throw new AfsException("No service found for key " + serviceKey + "");
+        ServiceExtension<T, U> serviceExtension = (ServiceExtension<T, U>) serviceExtensions.get(serviceKey);
+        if (serviceExtension == null) {
+            throw new AfsException("No service extension found for key " + serviceKey + "");
```

Comment:
Remove the quotes at the end of this message

Reply:
Done